### PR TITLE
`azurerm_kubernetes_cluster` - support for the `web_app_routing` property

### PR DESCRIPTION
**Pull request by ms-henglu**
```
=== RUN   TestAccKubernetesCluster_webAppRouting
=== PAUSE TestAccKubernetesCluster_webAppRouting
=== CONT  TestAccKubernetesCluster_webAppRouting
--- PASS: TestAccKubernetesCluster_webAppRouting (1047.24s)
PASS
```

### CODE REVIEW

File: internal/services/containers/kubernetes_cluster_other_resource_test.go [hunk 1]
```diff
@@ -2104,3 +2126,72 @@ resource "azurerm_kubernetes_cluster" "test" {
...
+    name       = "default"
+    node_count = 1
+    vm_size    = "Standard_DS2_v2"
+  }
+  identity {
+    type = "SystemAssigned"
+  }
+  web_app_routing {
+    dns_zone_id = azurerm_dns_zone.test.id
+  }
```
Comment (stephybun) [inline internal/services/containers/kubernetes_cluster_other_resource_test.go:2161]:
```suggestion
  
  default_node_pool {
    name       = "default"
    node_count = 1
    vm_size    = "Standard_DS2_v2"
  }
  
  identity {
    type = "SystemAssigned"
  }
  
  web_app_routing {
    dns_zone_id = azurerm_dns_zone.test.id
  }
```

File: internal/services/containers/kubernetes_cluster_other_resource_test.go [hunk 2]
```diff
@@ -2104,3 +2126,72 @@ resource "azurerm_kubernetes_cluster" "test" {
...
+  resource_group_name = azurerm_resource_group.test.name
+  dns_prefix          = "acctestaks%d"
+  default_node_pool {
+    name       = "default"
+    node_count = 1
+    vm_size    = "Standard_DS2_v2"
+  }
+  identity {
+    type = "SystemAssigned"
+  }
```
Comment (stephybun) [inline internal/services/containers/kubernetes_cluster_other_resource_test.go:2194]:
```suggestion
  
  default_node_pool {
    name       = "default"
    node_count = 1
    vm_size    = "Standard_DS2_v2"
  }
  
  identity {
    type = "SystemAssigned"
  }
```

File: website/docs/r/kubernetes_cluster.html.markdown [hunk 3]
```diff
@@ -682,6 +684,12 @@ A `sysctl_config` block supports the following:
 
 ---
 
+A `web_app_routing` block supports the following:
+
+* `dns_zone_id` - (Required) Specifies the ID of the DNS Zone to be associated with the web app when Web App Routing is enabled.
```
Comment (stephybun) [inline website/docs/r/kubernetes_cluster.html.markdown:689]: Suggestion on wording and explanation of the property

```suggestion
* `dns_zone_id` - (Required) Specifies the ID of the DNS Zone in which DNS entries are created for applications deployed to the cluster when Web App Routing is enabled.


```

**Review by stephybun — COMMENTED** [review]
Thanks for this @ms-henglu, a few minor comments, but once they're resolved this should be good to go!

File: internal/services/containers/kubernetes_cluster_resource.go [hunk 4]
```diff
@@ -376,6 +377,21 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
...
 
+			"web_app_routing": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"dns_zone_id": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
```
Comment (stephybun) [inline internal/services/containers/kubernetes_cluster_resource.go:388]: Is it a valid use case/configuration for this to be left blank?

Comment (ms-henglu) [reply internal/services/containers/kubernetes_cluster_resource.go:388]: Yes, I've confirmed with service team.

Comment (stephybun) [reply internal/services/containers/kubernetes_cluster_resource.go:388]: If it's omitted then the dns_zone_id is set to an empty string "". Where are the DNS entries created for applications that are deployed to the cluster?

Comment (ms-henglu) [reply internal/services/containers/kubernetes_cluster_resource.go:388]: From service team:
> Probably no dns entries created automatically.

Comment (stephybun) [reply internal/services/containers/kubernetes_cluster_resource.go:388]: Then this should be marked as Required.

**Review by stephybun — CHANGES_REQUESTED** [review]

**Review by stephybun — APPROVED** [review]
Thanks @ms-henglu LGTM 🥟

**Pull request merged**